src/inverse-kinematics/include/iDynTree/InverseKinematics.h

```diff
@@ -640,7 +687,9 @@ class iDynTree::InverseKinematics
      * @return A constant reference to iDynTree::Model used by the inverse kinematics.
      */
     const Model & model() const;
```

Comment:
It is not clear here whether you get the full or the reduced model. In my opinion, by default it should return the reduced model, since it is what you expect from outside. Imaging to call the ``getNrOfDOFs`` method, I would expect to get the reduced number of DoFs, for example. This indeed creates compatibility problems with other projects.